nasa/cmr-graphql
# DAS-1900: As a developer, I would like to be able to make an HTTPS request that is routed to my variable generation service, so that I can request UMM-Var records are generated in the cloud

# Overview

### What is the feature?
DAS-1900: As a developer, I would like to be able to make an HTTPS request that is routed to my variable generation service, so that I can request UMM-Var records are generated in the cloud

We decided that variable generation(earthdata-varinfo) will be routed through graphQL

### What is the Solution?

I created an earthdata-varinfo lambda. Also modified the cmr-graphQL to accept a generateVariableDrafts query under the existing collection query, which will return variable drafts generated by the earthdata-varinfo lambda.

### What areas of the application does this impact?

Collection query will have a new field(generateVariableDrafts) that can be requested.
Also added a new field for variableDrafts called metadataSpecification.

# Testing

### Reproduction steps

- **Environment for testing:**
- cmr-graphql
- **Collection to test with:**
- C1598621093-GES_DISC or similar

1. Navigate to Apollo Studio for the environment you will be working in
2. Request the following collection query: 
```
query Collection($params: CollectionInput) {
  collection(params: $params) {
    conceptId
    generateVariableDrafts {
      count
      items {
        dataType
        definition
        dimensions
        longName
        name
        standardName
        units
        metadataSpecification
      }
    }
  }
}
```
4. Specify the variable input params conceptId ie: 
```
{
  "params": {
    "conceptId": "C1598621093-GES_DISC"
  }
}
```
5. Put in your authorization token in the Headers area.
6. Submit the query and you should receive something like the following back:
```
{
  "data": {
    "collection": {
      "conceptId": "C1598621093-GES_DISC",
      "generateVariableDrafts": {
        "count": 16,
        "items": [
          {
            "dataType": "int32",
            "definition": "Grid/time",
            "dimensions": [
              {
                "Name": "Grid/time",
                "Size": 1,
                "Type": "TIME_DIMENSION"
              }
            ],
            "longName": "Grid/time",
            "name": "Grid/time",
            "standardName": "time",
            "units": "seconds since 1970-01-01 00:00:00 UTC",
            "metadataSpecification": {
              "URL": "https://cdn.earthdata.nasa.gov/umm/variable/v1.8.2",
              "Name": "UMM-Var",
              "Version": "1.8.2"
            }
          },...
}
```


### Attachments
<img width="1496" alt="Screenshot 2023-08-31 at 11 54 15 PM" src="https://github.com/nasa/cmr-graphql/assets/51178394/fa0e2f90-4a48-4bbb-91ea-d25d3863dc43">

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist
